test/compilable/fix20964.d

```diff
@@ -0,0 +1,12 @@
+// https://issues.dlang.org/show_bug.cgi?id=20964
+
+uint foo(uint[3] m)
+{
+    auto p = m.ptr;
+    p += 3; // ok, p refers to the next right element after the block!
+    p -= 4; // ok, p refers to the next left element before the block!
```

Comment:
Are there tests for indexing outside the bounds of the array that's being pointed at?

Reply:
No idea. I have added the test for issue.